agent/xds/builtinextensiontemplate/template.go

```diff
@@ -166,6 +166,8 @@ func (envoyExtension EnvoyExtension) patchTerminatingGatewayListener(config xdsc
 			if ok {
 				filters = append(filters, newFilter)
 				patched = true
+			} else {
+				filters = append(filters, filter)
```

Comment:
If `err != nil` then we'll be calling `filters = append(filters, filter)` twice, right?
I think the `err`+`ok` check needs to be more robust for all cases here to prevent possible bugs